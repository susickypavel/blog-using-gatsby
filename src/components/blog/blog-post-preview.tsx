import React from "react"

import { MdxFrontmatter } from "@generated/graphql"
import { Link } from "gatsby"

interface Props {
  frontmatter: MdxFrontmatter
  slug: string
}

const BlogPostPreview: React.FC<Props> = ({ frontmatter: { title, date }, slug }) => {
  return (
    <div>
      <h1>{title}</h1>
      <time>{date}</time>
      <Link to={slug}>Read more...</Link>
    </div>
  )
}

export default BlogPostPreview
