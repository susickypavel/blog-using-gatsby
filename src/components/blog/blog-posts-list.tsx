import React from "react"

import BlogPostPreview from "./blog-post-preview"
import { MdxEdge } from "@generated/graphql"
import { PageTitle } from "@css/global-styles"

interface Props {
  edges: MdxEdge[]
  header?: boolean
}

const BlogPostsList: React.FC<Props> = ({ edges, header = true }) => {
  return (
    <div>
      {header ? <PageTitle>ARTICLES [{edges.length}]</PageTitle> : null}
      {edges.map(({ node }) => (
        <BlogPostPreview node={node} key={node.id} />
      ))}
    </div>
  )
}

export default BlogPostsList
