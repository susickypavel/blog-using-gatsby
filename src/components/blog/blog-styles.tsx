import styled from "@utils/styled"
import { Link } from "gatsby"
import {
  LAYOUT_MEDIA_QUERY_BP,
  BLOG_HOLDER_PADDING,
  BLOG_HOLDER_PADDING_SMALL
} from "@css/constants"
import { PageTitle } from "@css/global-styles"

export const BlogPostTitle = styled(PageTitle)`
  font-size: 3.2rem;
  text-align: center;

  padding: 24px;

  @media (max-width: ${LAYOUT_MEDIA_QUERY_BP}) {
    padding: 16px;
  }

  @media (max-width: 425px) {
    font-size: 1.6rem;
  }
`

export const BlogPostPreviewWrapper = styled(Link)`
  display: block;
  text-decoration: none;
  text-align: center;

  padding: 8px 16px;
  color: ${props => props.theme.color.primary};

  & h3 {
    font-size: 2rem;
    font-weight: lighter;
  }

  &:nth-of-type(even) {
  }

  &:nth-of-type(even) {
    background-color: ${props => props.theme.bg.secondary};
  }

  &:hover {
    text-decoration: underline;
  }
`

export const TagsHolder = styled.div`
  text-align: left;
  margin: 0px 24px 24px 24px;

  @media (max-width: ${LAYOUT_MEDIA_QUERY_BP}) {
    margin: 0 16px 16px 16px;
  }
`
// TODO: Extract to constants
export const TagWrapper = styled(Link)<{
  colors: {
    background: string
    foreground: string
  }
}>`
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25), 0px 0px 4px rgba(0, 0, 0, 0.25);
  background-color: ${props => props.colors.background};
  color: ${props => props.colors.foreground};
  display: inline-block;
  border-radius: 4px;
  padding: 8px 12px;
  margin: 3px;
  font-size: 1.4rem;
  font-weight: bold;
  text-decoration: none;
`

export const BlogContentHolder = styled.div`
  padding: ${BLOG_HOLDER_PADDING};
  padding-top: 0 !important;
  font-size: 2rem;

  @media only screen and (max-width: ${LAYOUT_MEDIA_QUERY_BP}) {
    padding: ${BLOG_HOLDER_PADDING_SMALL};
  }

  & *:first-child {
    margin-top: 0 !important;
  }

  & a {
    color: ${props => props.theme.color.links};
  }

  @media (max-width: 425px) {
    font-size: 1.6rem;
  }
`

export const BlogTagsPageTitle = styled.h1`
  text-align: center;
  margin: 16px 0;
`

export const BlogTagsGoBack = styled(Link)`
  display: block;
  margin: 0 auto;
  width: 100px;
  text-align: center;
  color: ${props => props.theme.color.links};
  font-size: 1.6rem;
  font-weight: bold;
  margin-bottom: 16px;
`
