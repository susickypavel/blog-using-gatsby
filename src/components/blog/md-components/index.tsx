import React from "react"
import styled from "@utils/styled"
import {
  BLOCKQUOTE_PADDING,
  BLOCKQUOTE_MARGIN,
  BLOCKQUOTE_BORDERRADIUS,
  BLOCKQUOTE_FONTSIZE
} from "@css/constants"

const BlockQuoteWrapper = styled.blockquote`
  background: ${props => props.theme.bg.secondary};
  padding: ${BLOCKQUOTE_PADDING};
  margin: ${BLOCKQUOTE_MARGIN};
  border-radius: ${BLOCKQUOTE_BORDERRADIUS};
  font-size: ${BLOCKQUOTE_FONTSIZE};

  border-left: 4px solid ${props => props.theme.color.primary};

  & p {
    margin: 0;
  }
`

export const UnorderedList = styled.ul`
  list-style-type: inherit;
  padding-left: 20px;
  margin-top: 8px;
  margin-left: 16px;

  & li {
    margin: 2px 0;
  }
`

export const OrderedList = styled.ol`
  list-style-type: decimal;
  padding-left: 20px;
  margin-top: 8px;
  margin-left: 16px;

  & li {
    margin: 2px 0;
  }
`

export const Heading: React.FC<{
  heading: "h1" | "h2" | "h3" | "h4" | "h5" | "h6"
  fontSize: number
}> = ({ heading, fontSize, children }) => {
  const HeadingComponent = styled(heading)`
    font-size: ${fontSize}rem;
    margin: 32px 0;
  `

  return <HeadingComponent>{children}</HeadingComponent>
}

export const BlockQuote: React.FC = ({ children }) => {
  return (
    <BlockQuoteWrapper>
      <p>{children}</p>
    </BlockQuoteWrapper>
  )
}

export const Paragraph = styled.p`
  line-height: 28px;
  margin: 8px 0;
`
