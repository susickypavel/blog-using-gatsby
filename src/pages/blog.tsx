import React from "react"
import { graphql } from "gatsby"

import PageLayout from "@components/PageLayout"
import BlogPostsList from "@components/blog/blog-posts-list"
import { MdxGroupConnection } from "@generated/graphql"

interface Props {
  data: {
    allMdx: MdxGroupConnection
  }
}

const Blog: React.FC<Props> = ({
  data: {
    allMdx: { edges }
  }
}) => {
  return (
    <PageLayout>
      <BlogPostsList edges={edges} />
    </PageLayout>
  )
}

export const pageQuery = graphql`
  query blogIndex {
    allMdx {
      edges {
        node {
          id
          excerpt
          frontmatter {
            title
          }
          fields {
            slug
          }
        }
      }
    }
  }
`

export default Blog
