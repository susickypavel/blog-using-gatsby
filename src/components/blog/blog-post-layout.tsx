import React from "react"
import { graphql } from "gatsby"
// TODO: Do not have typings
// @ts-ignore
import { MDXRenderer } from "gatsby-plugin-mdx"

import PageLayout from "@components/PageLayout"
import { BlogPostListTitle, BlogContentHolder } from "./blog-styles"

const PageTemplate: React.FC<any> = ({ data: { mdx } }) => {
  return (
    <PageLayout title={mdx.frontmatter.title}>
      <BlogPostListTitle>{mdx.frontmatter.title}</BlogPostListTitle>
      <BlogContentHolder>
        <MDXRenderer>{mdx.body}</MDXRenderer>
      </BlogContentHolder>
    </PageLayout>
  )
}

export const pageQuery = graphql`
  query BlogPostQuery($id: String) {
    mdx(id: { eq: $id }) {
      id
      body
      frontmatter {
        title
      }
    }
  }
`

export default PageTemplate
