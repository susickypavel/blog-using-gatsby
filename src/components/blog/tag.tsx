import React from "react"
import { TagWrapper } from "./blog-styles"

interface Props {
  name: string
}

const Tag: React.FC<Props> = ({ name }) => {
  const colors: any = {
    javascript: { background: "#F0DB4F", foreground: "#323330" },
    typescript: { background: "#007acc", foreground: "white" },
    react: { background: "#61DBFB", foreground: "black" },
    angular: { background: "#b52e31", foreground: "white" },
    vue: { background: "#41B883", foreground: "#34495E" }
  }

  return (
    <TagWrapper
      to="/tags"
      state={{ tag: name }}
      colors={colors[name] || { background: "gray", foreground: "white" }}
    >
      {name}
    </TagWrapper>
  )
}

export default Tag
