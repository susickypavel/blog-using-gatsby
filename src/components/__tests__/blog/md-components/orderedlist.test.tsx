import React from "react"
import serializer from "jest-emotion"
import { render } from "@testing-library/react"

import { OrderedList } from "@components/blog/md-components"

expect.addSnapshotSerializer(serializer)

describe("Ordered List component", () => {
  test("should render", () => {
    const { container } = render(
      <OrderedList>
        <li>Test</li>
      </OrderedList>
    )

    expect(container).toMatchSnapshot()
  })
})
