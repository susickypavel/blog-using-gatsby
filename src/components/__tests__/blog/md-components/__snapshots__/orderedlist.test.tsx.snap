// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ordered List component should render 1`] = `
.emotion-0 {
  list-style-type: decimal;
  padding-left: 20px;
  margin-top: 8px;
  margin-left: 16px;
}

.emotion-0 li {
  margin: 2px 0;
}

<div>
  <ol
    class="emotion-0"
  >
    <li>
      Test
    </li>
  </ol>
</div>
`;
